### PR TITLE
Develop dan

### PR DESCRIPTION
**Pull request by Moziki**
Created sprocs to update dependencies in db.
Connected methods from front end to back end for dependencies

### CODE REVIEW

File: WPF/Model/Task.cs [hunk 1]
```diff
@@ -184,17 +184,37 @@ public override bool RemoveWorker(User worker)
...
         public void UpdateDependencies()
         {
-
+            
         }
-
+        //Creates a dependency, the parameter is the Root, and the current task is the dependent. 
         public void AddDependency(Task dependency)
         {
-
```
Comment (Robert-N7) [inline WPF/Model/Task.cs:192]: Need to add to list of dependencies here.

File: WPF/Model/Task.cs [hunk 2]
```diff
@@ -184,17 +184,37 @@ public override bool RemoveWorker(User worker)
...
+            string query = "EXEC CreateDependency @RootId, @DependentId";
```
Comment (Robert-N7) [inline WPF/Model/Task.cs:192]: The database logic should be in its own method.

File: WPF/Model/Task.cs [hunk 3]
```diff
@@ -184,17 +184,37 @@ public override bool RemoveWorker(User worker)
...
+            SqlCommand command = OpenConnection(query);
+            command.Parameters.AddWithValue("@RootId", dependency.Id);
+            command.Parameters.AddWithValue("@DependentId", this.Id);
+            command.ExecuteNonQuery();
+            CloseConnection();
```
Comment (Robert-N7) [inline WPF/Model/Task.cs:197]: And remember to call NotifyUpdate() to update subscribers that a change happened

File: WPF/Model/Task.cs [hunk 4]
```diff
@@ -184,17 +184,37 @@ public override bool RemoveWorker(User worker)
...
         }
-
+        //Removes a dependency, the parameter is the Root, and the current task is the dependent. 
         public void RemoveDependency(Task dependency)
         {
-
```
Comment (Robert-N7) [inline WPF/Model/Task.cs:197]: Same as above, update dependencies list

File: WPF/Model/Task.cs [hunk 5]
```diff
@@ -184,17 +184,37 @@ public override bool RemoveWorker(User worker)
...
+            string query = "EXEC RemoveDependency @RootId, @DependentId";
+            SqlCommand command = OpenConnection(query);
+            command.Parameters.AddWithValue("@RootId", dependency.Id);
+            command.Parameters.AddWithValue("@DependentId", this.Id);
+            command.ExecuteNonQuery();
+            CloseConnection();
+        }
```
Comment (Robert-N7) [inline WPF/Model/Task.cs:208]: And call NotifyUpdate()

File: WPF/View/Controls/TaskControl.xaml.cs [hunk 6]
```diff
@@ -163,6 +163,8 @@ private void mi_editTask_Click(object sender, RoutedEventArgs e)
 
         private void mi_deleteTask_Click(object sender, RoutedEventArgs e)
         {
+            Task.DeleteDependency(Task);
```
Comment (Robert-N7) [inline WPF/View/Controls/TaskControl.xaml.cs:166]: This should be handled in the model

File: WPF/Model/Task.cs [hunk 7]
```diff
@@ -184,17 +184,37 @@ public override bool RemoveWorker(User worker)
...
+        //Calls sproc DeleteDependency that removes all dependencies associated with the task to be deleted
+        public void DeleteDependency(Task delTask)
+        {
```
Comment (Robert-N7) [inline WPF/Model/Task.cs:211]: Also update in the project's tasks

File: WPF/View/Pages/WorkSpace.xaml.cs [hunk 8]
```diff
@@ -101,9 +101,14 @@ private void AddDependencies()
             foreach(KeyValuePair<Task, TaskControl> keyValue in taskControls)
             {
                 foreach (Task t in keyValue.Key.Dependencies)
-                    keyValue.Value.ConnectDependentControl(taskControls[t]);
+                    if (taskControls.ContainsKey(t))
+                        keyValue.Value.ConnectDependentControl(taskControls[t]);
             }
         }
+        private void RemoveDependencies(Task t)
```
Comment (Robert-N7) [inline WPF/View/Pages/WorkSpace.xaml.cs:108]: No

File: PertDB/dbo/StoredProcedures/DeleteDependency.sql [hunk 9]
```diff
@@ -0,0 +1,7 @@
+﻿CREATE PROCEDURE [dbo].[DeleteDependency]
+	@taskId int
+AS
+BEGIN
+	DELETE FROM [dbo].Dependency
+	WHERE RootId = @taskId OR DependentId = @taskId
```
Comment (Robert-N7) [inline PertDB/dbo/StoredProcedures/DeleteDependency.sql:6]: This is deleting all incoming and outgoing dependencies? This is already handled in the DeleteTask trigger.

**Review by Robert-N7 — COMMENTED** [review]
In general, we need to either keep the model's state updated and notify subscribers, or ping the dbreader to refresh after a database read/write

File: PertDB/dbo/StoredProcedures/DeleteDependency.sql (hunk 9 above)
Comment (Moziki) [reply PertDB/dbo/StoredProcedures/DeleteDependency.sql:6]: for potential cases when we aren't necessarily deleting a task but want to remove all dependencies associated with it

File: WPF/View/Pages/WorkSpace.xaml.cs (hunk 8 above)
Comment (Moziki) [reply WPF/View/Pages/WorkSpace.xaml.cs:108]: Yes?

**Review by Robert-N7 — APPROVED** [review]

**Pull request merged**